src/app/ui/cpwsarea/wsnewmap/wsnewmap.go

```diff
@@ -0,0 +1,199 @@
+package wsnewmap
+
+import (
+	"log"
+	"math"
+	"path/filepath"
+	"sdmm/app/command"
+	"sdmm/app/prefs"
+	"sdmm/app/ui/cpwsarea/workspace"
+	"sdmm/app/window"
+	"sdmm/dmapi/dmenv"
+	"sdmm/imguiext"
+	"sdmm/imguiext/icon"
+	"sdmm/imguiext/markdown"
+
+	"github.com/SpaiR/imgui-go"
+	"github.com/sqweek/dialog"
+)
+
+type App interface {
+	LoadedEnvironment() *dmenv.Dme
+
+	CommandStorage() *command.Storage
+	Prefs() prefs.Prefs
+
+	DoOpenMapByPath(path string)
+	DoClose()
+}
+
+type WsNewMap struct {
+	workspace.Content
+
+	app App
+
+	mapWidth  int // x
+	mapHeight int // y
+	mapZDepth int // z
+
+	multiZType string
+
+	newMapMetaFile string
+}
+
+func New(app App) *WsNewMap {
+	return &WsNewMap{
+		app: app,
+
+		mapWidth:       1,
+		mapHeight:      1,
+		mapZDepth:      1,
+		multiZType:     Default,
+		newMapMetaFile: "",
+	}
+}
+
+func (ws *WsNewMap) Name() string {
+	return icon.FaPlus + " New Map"
+}
+
+func (ws *WsNewMap) Title() string {
+	return "New Map"
+}
+
+func (ws *WsNewMap) Process() {
+	ws.showContent()
+}
+
+func (ws *WsNewMap) showContent() {
```

Comment:
To organize the code better, I can recommend to split every "part" of the workspace to separate method. So you can have something like:
```
func (ws *WsNewMap) showContent() {
    showHeader()
    showInputWidth()
    showInputHeight()
    ...etc...
}
```
Thus the code will be readable.

Also, the process itself can be moved to a separate file as well.